Fix particles freezing by resetting particle timer

GanidhuAbey opened the pull request:
one line change, resets particle lifetime when new particle is emitted. closes #56886

Calinou commented on servers/rendering/renderer_rd/renderer_storage_rd.cpp:
```diff
@@ -4825,6 +4825,7 @@ void RendererStorageRD::particles_request_process(RID p_particles) {
 
 	if (!particles->dirty) {
 		particles->dirty = true;
+        	particles->inactive_time = 0;
```
Indentation should use tabs, not spaces:

```suggestion
		particles->inactive_time = 0;
```

Calinou commented:
Thanks for opening a pull request!

Please [amend the commit to use a human-readable commit message](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-messages-with-readability-in-mind) (`git commit --amend`). This way, people reading the Git log can easily figure out what's going on :slightly_smiling_face:

Once this is done, please squash the commits together so we can merge this. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for instructions :slightly_smiling_face:

GanidhuAbey commented:
done, thank you

reduz commented on servers/rendering/renderer_rd/renderer_storage_rd.cpp:
```diff
@@ -4825,6 +4825,7 @@ void RendererStorageRD::particles_request_process(RID p_particles) {
 
 	if (!particles->dirty) {
 		particles->dirty = true;
+		particles->inactive_time = 0;
```
This change will just make it impossible for particles to become inactive as long as they are visible, so definitely not a correct fix. The main issue is that I just dont understand the problem and cant see anything in the reproduction project of the original issue.

GanidhuAbey replied:
Hi,

Yea I definitely should have looked more into how that function was being called, I assumed that it was only being called when the a new particle was instanced and turns out, that's definitely wrong. The problem is, I am not seeing the behavior you're describing from this fix, the particle lasts only until its lifetime and disappears.

Any idea why this is happening?

clayjohn reviewed (changes requested):
This appears to be the correct fix to me. 

I guess the issue was coming from a single frame where the particles would get set as inactive before this function made them active again in the following frame. 

It makes sense that `inactive_time`` is reset after each emission as that particle should be allowed to stay active for its lifetime.

Before merging, you will need to delete the print statement you added and squash your commits together. The whole thing should contain just a single commit (for instructions on how to do so, see here: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase)

GanidhuAbey committed:
Fix particle system from going inactive early

Prevents particles from freezing and dissapearing by resetting
it's inactive time when particles are emitted.

clayjohn reviewed (approved):
Thanks! Everything is looking good now

GanidhuAbey commented:
Cool, it took a lot of time to learn how to use git properly, haha.

It's exactly as you said, the system was getting set to inactive too early, I figured that the best way to deal with this was resetting the inactive_time and letting the new particle survive it's entire lifetime.

Thanks for all your help in this, had a huge breakthrough with the problem while talking with you about it :)

akien-mga commented:
Thanks!
